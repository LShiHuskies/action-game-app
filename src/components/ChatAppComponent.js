import React, { Fragment } from 'react';
import ChatAppMessage from './ChatAppMessage';


const ChatAppComponent = ({ messages, user }) => (
    <Fragment>
       {messages.map(message => <ChatAppMessage messageObj={message} userObj={user} />)}
    </Fragment>
);


export default ChatAppComponent;

//    const { user_id, created_at } = message;
//     if (user_id === user.id) {
//         return (
//         <div className="message-orange" key={created_at}>
//           <p className="message-content">{message.message}</p>
//           <div className="message-timestamp-right"><strong>{message.user.username}</strong> {new Date(created_at).toLocaleTimeString()}</div>
//         </div>
//         )
//     } else {
//         return (
//         <div className="message-blue" key={created_at}>
//         <p className="message-content">{message.message}</p>
//         <div className="message-timestamp-left"><strong>{message.user.username}</strong> {new Date(created_at).toLocaleTimeString()}</div>
//     </div>
//     )}
