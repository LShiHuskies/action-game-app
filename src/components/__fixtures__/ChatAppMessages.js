const ChatAppMessages = [
        {
            "user_id": 45,
            "chatroom_id": 3,
            "message": "How are you?",
            "created_at": "2022-03-17T04:26:29.123Z",
            "updated_at": "2022-03-17T04:26:29.123Z",
            "user": {
                "id": 45,
                "username": "FirstUserEver",
                "first_name": "FirstName",
                "last_name": "Surname",
                "email": "*********@gmail.com",
                "password_digest": "_____________________",
                "created_at": "2022-03-04T04:27:25.534Z",
                "updated_at": "2022-03-04T04:27:25.534Z",
                "recovery_question": null,
                "recovery_answer_digest": null,
                "reset_digest": null,
                "reset_sent_at": null,
                "admin": false,
                "activation_digest": "_____________________________________-",
                "activated": true,
                "activated_at": "2022-03-04T05:27:37.990Z"
            },
            "chatroom": {
                "id": 3,
                "user_game_id": 43,
                "created_at": "2022-03-05T03:39:37.794Z",
                "updated_at": "2022-03-05T03:39:37.794Z"
            }
        },
        {
            "user_id": 45,
            "chatroom_id": 3,
            "message": "The Best in the House",
            "created_at": "2022-03-17T04:26:32.352Z",
            "updated_at": "2022-03-17T04:26:32.352Z",
            "user": {
                "id": 45,
                "username": "FirstUserEver",
                "first_name": "FirstName",
                "last_name": "Surname",
                "email": "*********@gmail.com",
                "password_digest": "_____________________",
                "created_at": "2022-03-04T04:27:25.534Z",
                "updated_at": "2022-03-04T04:27:25.534Z",
                "recovery_question": null,
                "recovery_answer_digest": null,
                "reset_digest": null,
                "reset_sent_at": null,
                "admin": false,
                "activation_digest": "_____________________________________-",
                "activated": true,
                "activated_at": "2022-03-04T05:27:37.990Z"
            },
            "chatroom": {
                "id": 3,
                "user_game_id": 43,
                "created_at": "2022-03-05T03:39:37.794Z",
                "updated_at": "2022-03-05T03:39:37.794Z"
            }
        },
        {
            "user_id": 45,
            "chatroom_id": 3,
            "message": "kdjsfksdjfds",
            "created_at": "2022-03-17T04:26:34.550Z",
            "updated_at": "2022-03-17T04:26:34.550Z",
            "user": {
                "id": 45,
                "username": "FirstUserEver",
                "first_name": "FirstName",
                "last_name": "Surname",
                "email": "*********@gmail.com",
                "password_digest": "_____________________",
                "created_at": "2022-03-04T04:27:25.534Z",
                "updated_at": "2022-03-04T04:27:25.534Z",
                "recovery_question": null,
                "recovery_answer_digest": null,
                "reset_digest": null,
                "reset_sent_at": null,
                "admin": false,
                "activation_digest": "_____________________________________-",
                "activated": true,
                "activated_at": "2022-03-04T05:27:37.990Z"
            },
            "chatroom": {
                "id": 3,
                "user_game_id": 43,
                "created_at": "2022-03-05T03:39:37.794Z",
                "updated_at": "2022-03-05T03:39:37.794Z"
            }
        },
        {
            "user_id": 42,
            "chatroom_id": 3,
            "message": "dfkjsdkfjsdklfjsdlfjds",
            "created_at": "2022-03-17T04:27:14.800Z",
            "updated_at": "2022-03-17T04:27:14.800Z",
            "user": {
                "id": 42,
                "username": "SecondUserEver",
                "first_name": "FirstName",
                "last_name": "Surname",
                "email": "*********@gmail.com",
                "password_digest": "_____________________",
                "created_at": "2022-03-04T04:27:25.534Z",
                "updated_at": "2022-03-04T04:27:25.534Z",
                "recovery_question": null,
                "recovery_answer_digest": null,
                "reset_digest": null,
                "reset_sent_at": null,
                "admin": false,
                "activation_digest": "_____________________________________-",
                "activated": true,
                "activated_at": "2022-03-04T05:27:37.990Z"
            },
            "chatroom": {
                "id": 3,
                "user_game_id": 43,
                "created_at": "2022-03-05T03:39:37.794Z",
                "updated_at": "2022-03-05T03:39:37.794Z"
            }
        },
        {
            "user_id": 43,
            "chatroom_id": 3,
            "message": "Another Message for this room",
            "created_at": "2022-03-17T04:27:14.800Z",
            "updated_at": "2022-03-17T04:27:14.800Z",
            "user": {
                "id": 43,
                "username": "ThirdUserEver",
                "first_name": "FirstName",
                "last_name": "Surname",
                "email": "*********@gmail.com",
                "password_digest": "_____________________",
                "created_at": "2022-03-04T04:27:25.534Z",
                "updated_at": "2022-03-04T04:27:25.534Z",
                "recovery_question": null,
                "recovery_answer_digest": null,
                "reset_digest": null,
                "reset_sent_at": null,
                "admin": false,
                "activation_digest": "_____________________________________-",
                "activated": true,
                "activated_at": "2022-03-04T05:27:37.990Z"
            },
            "chatroom": {
                "id": 3,
                "user_game_id": 43,
                "created_at": "2022-03-05T03:39:37.794Z",
                "updated_at": "2022-03-05T03:39:37.794Z"
            }
        }
    ];

export default ChatAppMessages;
