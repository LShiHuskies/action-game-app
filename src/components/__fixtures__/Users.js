export const user1 = {
        "id": 45,
        "username": "FirstUserEver",
        "first_name": "FirstName",
        "last_name": "Surname",
        "email": "*********@gmail.com",
        "password_digest": "_____________________",
        "created_at": "2022-03-04T04:27:25.534Z",
        "updated_at": "2022-03-04T04:27:25.534Z",
        "recovery_question": null,
        "recovery_answer_digest": null,
        "reset_digest": null,
        "reset_sent_at": null,
        "admin": false,
        "activation_digest": "_____________________________________-",
        "activated": true,
        "activated_at": "2022-03-04T05:27:37.990Z"
    };

export const user2 = {
    "id": 42,
    "username": "SecondUserEver",
    "first_name": "FirstName",
    "last_name": "Surname",
    "email": "*********@gmail.com",
    "password_digest": "_____________________",
    "created_at": "2022-03-04T04:27:25.534Z",
    "updated_at": "2022-03-04T04:27:25.534Z",
    "recovery_question": null,
    "recovery_answer_digest": null,
    "reset_digest": null,
    "reset_sent_at": null,
    "admin": false,
    "activation_digest": "_____________________________________-",
    "activated": true,
    "activated_at": "2022-03-04T05:27:37.990Z"
};


export const user3 = {
    "id": 43,
    "username": "ThirdUserEver",
    "first_name": "FirstName",
    "last_name": "Surname",
    "email": "*********@gmail.com",
    "password_digest": "_____________________",
    "created_at": "2022-03-04T04:27:25.534Z",
    "updated_at": "2022-03-04T04:27:25.534Z",
    "recovery_question": null,
    "recovery_answer_digest": null,
    "reset_digest": null,
    "reset_sent_at": null,
    "admin": false,
    "activation_digest": "_____________________________________-",
    "activated": true,
    "activated_at": "2022-03-04T05:27:37.990Z"
};
