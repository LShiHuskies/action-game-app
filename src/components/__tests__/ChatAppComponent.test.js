import React from 'react';
import renderer from 'react-test-renderer';
import ChatAppMessages from '../__fixtures__/ChatAppMessages';
import { user1 } from '../__fixtures__/Users';

import ChatAppComponent from '../ChatAppComponent';





describe('<ChatAppComponenet />', () => {


    it('should match snapshot', () => {
        const tree = renderer.create(<ChatAppComponent messages={ChatAppMessages} user={user1} />);

        expect(tree.toJSON()).toMatchSnapshot();
    });

});