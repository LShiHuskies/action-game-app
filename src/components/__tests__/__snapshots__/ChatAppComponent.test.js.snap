// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatAppComponenet /> should match snapshot 1`] = `
Array [
  <div
    className="message-orange"
  >
    <p
      className="message-content"
    >
      How are you?
    </p>
    <div
      className="message-timestamp-right"
    >
      <strong>
        FirstUserEver
      </strong>
       
      12:26:29 AM
    </div>
  </div>,
  <div
    className="message-orange"
  >
    <p
      className="message-content"
    >
      The Best in the House
    </p>
    <div
      className="message-timestamp-right"
    >
      <strong>
        FirstUserEver
      </strong>
       
      12:26:32 AM
    </div>
  </div>,
  <div
    className="message-orange"
  >
    <p
      className="message-content"
    >
      kdjsfksdjfds
    </p>
    <div
      className="message-timestamp-right"
    >
      <strong>
        FirstUserEver
      </strong>
       
      12:26:34 AM
    </div>
  </div>,
  <div
    className="message-blue"
  >
    <p
      className="message-content"
    >
      dfkjsdkfjsdklfjsdlfjds
    </p>
    <div
      className="message-timestamp-left"
    >
      <strong>
        SecondUserEver
      </strong>
       
      12:27:14 AM
    </div>
  </div>,
  <div
    className="message-blue"
  >
    <p
      className="message-content"
    >
      Another Message for this room
    </p>
    <div
      className="message-timestamp-left"
    >
      <strong>
        ThirdUserEver
      </strong>
       
      12:27:14 AM
    </div>
  </div>,
]
`;
